### PR TITLE
Massive bump

### PR DESCRIPTION
**Pull request by hritikaggarwal2**
- added CSS to Login and SignUp
- Router only shows login/signup/landing if you are not logged in
- Changed New User to Sign Up to avoid confusion
- Added a regex which doesn’t give linter issues
- refactored code
- fixed bugs with landing page
- fixed user provider to have only a single user (did to avoid confusion between firebase user and husky user)
- added style to create new group
- added style to ValidateInput
- Removed redundant code from ValidateInput
- restructured pop form to become generic
- renamed CreateGroup to CreateGroupPopUp
- Added onClick functionality to user groups with sending data
- Made the user groups work for not just one chat but for every chat (I might have messed this somehow so I fixed that)
- Also used ChatClass instead for consistency and added ownerId in chatClass
- added auto update to fetch calls so the snapshot gets updated whenever there is a change in data. So, no need to refresh to see chat
- and more…

### CODE REVIEW

File: web/src/components/Chat.js [hunk 1]
```diff
@@ -57,19 +57,21 @@ export default function Chat(props) {
...
           user.display_name,
-          user.uid,
+          user.uwid,
           firebase.firestore.Timestamp.now()
         )
-      )
-      .then((docRef) => {
+      );
+      // NOT SURE WHY THE CODE BELOW IS COMMENTED OUT
+      // DO WE NEED IT?
```
[the remark this answers is not in the document]
Comment (hritikaggarwal2) [reply web/src/components/Chat.js:65]: yeah, so the build is passing. It's just the test that was failing. That test is probably no longer valid

File: web/src/components/Chat.js [hunk 2]
```diff
@@ -57,19 +57,21 @@ export default function Chat(props) {
         new ChatClass(
           outMessage,
           user.display_name,
-          user.uid,
+          user.uwid,
```
[the remark this answers is not in the document]
Comment (ilyasrepo) [reply web/src/components/Chat.js:60]: Throws an error if I change it back to uid

[the remark this answers is not in the document]
Comment (hritikaggarwal2) [reply web/src/components/Chat.js:60]: weird it was working for me :/

**Review by ilyasrepo — APPROVED** [review]
Reviewed on local machine. I made changes and temporarily removed a test to pass the build. Everything looks good.

**Pull request merged**